tests/classes/test_function.py

```diff
@@ -379,3 +379,161 @@ def test_maximal_simplices(edgelist5, edgelist8):
     H = xgi.Hypergraph(edgelist5)
     with pytest.raises(XGIError):
         xgi.maximal_simplices(H)
+
+
+def test_density_empty():
+    tol = 1e-10
+    H = xgi.Hypergraph()
+    assert xgi.density(H) < tol
+    assert xgi.density(H, order=3) < tol
+    assert xgi.density(H, max_order=7) < tol
+    assert xgi.density(H, ignore_singletons=True) < tol
+
+
+def test_density_one_node():
+    tol = 1e-10
+    H = xgi.Hypergraph([[0]])  # one node, one singleton edge
+    assert abs(xgi.density(H) - 1) < tol
+    assert xgi.density(H, ignore_singletons=True) < tol
+
+    H = xgi.Hypergraph()
+    H.add_node(0)  # one node, no edges
+    assert xgi.density(H) < tol
+    assert xgi.density(H, order=2) < tol
+    assert xgi.density(H, max_order=13) < tol
+    assert xgi.density(H, ignore_singletons=True) < tol
+
+
+def test_density_two_nodes():
+    tol = 1e-10
+
+    H = xgi.Hypergraph()  # two nodes, no edges
+    H.add_nodes_from([0, 1])
+    assert xgi.density(H) < tol
+    assert xgi.density(H, order=3) < tol
+    assert xgi.density(H, max_order=6) < tol
+    assert xgi.density(H, ignore_singletons=True) < tol
+
+    H = xgi.Hypergraph()  # two nodes, one singleton edge
+    H.add_nodes_from([0, 1])
+    H.add_edge([0])
+    assert abs(xgi.density(H) - 1 / 3) < tol
+    assert abs(xgi.density(H, order=0) - 1 / 2) < tol
+    assert xgi.density(H, order=1) < tol
+    assert xgi.density(H, ignore_singletons=True) < tol
+
+    H = xgi.Hypergraph()  # two nodes, the other singleton edge
```

Comment:
Is this case necessary? It seems symmetric with the previous one

Reply:
I guess not, deleted!